### PR TITLE
OpenMPTarget: FunctorAdapter to centralize tag evaluation.

### PR DESCRIPTION
**Pull request by rgayatri23**
The PR centralizes the evaluation of a functor tag in a separate file.

### CODE REVIEW

**Comment by dalg24** [comment]
Retest this please

File: core/src/OpenMPTarget/Kokkos_OpenMPTarget_ParallelFor_Range.hpp [hunk 1]
```diff
@@ -50,15 +49,11 @@ class ParallelFor<FunctorType, Kokkos::RangePolicy<Traits...>,
 
     if (end <= begin) return;
 
-    FunctorType a_functor(m_functor);
+    auto const a_functor(m_functor);
```
[the remark this answers is not in the document]
Comment (rgayatri23) [reply core/src/OpenMPTarget/Kokkos_OpenMPTarget_ParallelFor_Range.hpp:52]: Yes, that's the reason for this style.

File: core/src/OpenMPTarget/Kokkos_OpenMPTarget_ParallelFor_MDRange.hpp [hunk 2]
```diff
@@ -55,13 +58,13 @@ class ParallelFor<FunctorType, Kokkos::MDRangePolicy<Traits...>,
...
     execute_tile<Policy::rank>(
-        unused, functor, policy,
+        unused, m_functor, policy,
         std::integral_constant<Iterate, Policy::inner_direction>());
   }
 
   template <int Rank>
   inline std::enable_if_t<Rank == 2> execute_tile(
-      typename Policy::point_type offset, const FunctorType& functor,
+      typename Policy::point_type offset, const FunctorAdapter& m_functor,
```
Comment (masterleinad) [inline core/src/OpenMPTarget/Kokkos_OpenMPTarget_ParallelFor_MDRange.hpp:67]:
```suggestion
      typename Policy::point_type offset, const FunctorAdapter& functor,
```
I would avoid shadowing the member variable.

**Review by masterleinad — APPROVED** [review]
Looks OK to me.

**Comment by dalg24** [comment]
The OpenMPTarget build passed

**Review by dalg24 — APPROVED** [review]

**Pull request merged**